tests/integration/HACKING.md

```diff
@@ -0,0 +1,468 @@
+
+```python
+setup(
+    name="my_custom_package",
+    ...
+	install_requires=[
+		"girder_worker",
+		....
+	],
+    extras_require={
+        'consumer':  ["complex_dependency", ...],
+    }
+	...
+)
+```
+
+This means the producer side code can be installed with pip as ```my_custom_package``` and the consumer side code (that includes the complex dependencies) can be installed as ```my_custom_package[consumer]```
+
+## Communicating with Girder from inside Tasks
+
+Tasks decorated with the Girder Worker application object have two main channels of interacting with the Girder instance that produced the task. 
+
+The first is the ```job_manager``` which provides an API for interacting with the specific Girder job model that is reflecting the state of Celery Task. The job_manger is an instance of ```girder_worker.utils.JobManager```. The job\_manager is available on the task object,  which means you must decorate the task with bind equal to true (e.g. ```@app.task(bind=true)```). This means the first argument to the task will be ```self``` and the job\_manger will be available as ```self.job_manager```.  For the most part,  task state transition is handled seamlessly through Girder Worker's use of [Celery Signals](http://docs.celeryproject.org/en/latest/userguide/signals.html). This means basic state transition (e.g.  running, success, failure, retry, etc) will be handled without any intervention within the task.  If you wish to use custom states,  update the Job progress, or to log specific information within your task,  you can do so using the this job\_manager. The job_manager is automatically generated regardless of whether you use the Traditional method or the Celery method of producing your task. The job\_manager is scoped so that it only has push access to the specific job it is assigned. If your task needs to leverage other components of the Girder REST API, including pushing data back to Girder or modifying Girder collections via the API,  then you should use the ```girder_client```
+
+The second channel for interacting with Girder from within a task is through the ```girder_client``` attribute.  Like the ```job_manager``` it is available on the Celery task object (so you must set bind equal to true to access it). The girder\_client is an instance of ```girder_client.GirderClient```. It provides an authenticated session for making REST requests against the Girder API. The authentication is handled through a token provided to the task through the reserved keyword ```girder_client_token```. This will allow the girder\_client within the task to act with the same permissions granted to the token. By default the girder\_client\_token is None,  meaning the girder\_client instance within the task will have the permissions of an unauthenticated user.
```

Comment:
`By default the girder\_client\_token is None` does this mean if you haven't set the token through the girder_client_token keyword arg?  If so, please clarify this.